runtime/drivers/bigquery/arrow.go

```diff
@@ -0,0 +1,155 @@
+package bigquery
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"sync/atomic"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/apache/arrow/go/v13/arrow"
+	"github.com/apache/arrow/go/v13/arrow/array"
+	"github.com/apache/arrow/go/v13/arrow/ipc"
+	"github.com/apache/arrow/go/v13/arrow/memory"
+	"github.com/rilldata/rill/runtime/pkg/observability"
+	"go.uber.org/zap"
+	"google.golang.org/api/iterator"
+)
+
+func (f *fileIterator) AsArrowRecordReader() (array.RecordReader, error) {
+	arrowIt, err := f.bqIter.ArrowIterator()
+	if err != nil {
+		return nil, err
+	}
+
+	allocator := memory.DefaultAllocator
+	buf := bytes.NewBuffer(arrowIt.SerializedArrowSchema())
+	rdr, err := ipc.NewReader(buf, ipc.WithAllocator(allocator))
+	if err != nil {
+		return nil, err
+	}
+	defer rdr.Release()
+
+	rec := &arrowRecordReader{
+		bqIter:      arrowIt,
+		arrowSchema: rdr.Schema(),
+		refCount:    1,
+		allocator:   allocator,
+		logger:      f.logger,
+		records:     make([]arrow.Record, 0),
+		ctx:         f.ctx,
+	}
+	return rec, nil
+}
+
+// some impl details are copied from array.simpleRecords
+type arrowRecordReader struct {
+	bqIter      bigquery.ArrowIterator
+	records     []arrow.Record
+	cur         arrow.Record
+	arrowSchema *arrow.Schema
+	refCount    int64
```

Comment:
Did you consider using an `atomic.Int64` instead? The docs seems to recommend that

Reply:
No I just copied from the `array.simpleRecords` code. But yeah thanks `atomic.Int64` is a better choice.